src/ngx_http_lua_headers_in.c

```diff
@@ -693,6 +768,22 @@ ngx_http_lua_set_input_header(ngx_http_request_t *r, ngx_str_t key,
     }
 #endif
 
+#endif
+    i = sizeof(ngx_http_lua_set_handlers)/sizeof(ngx_http_lua_set_header_t) - 1;
+    if (key.len < 64) {
+        ngx_strlow(&uc[0], key.data, key.len);
+        v = ngx_hash_find(lua_req_header_hash.hash, hv.hash, &uc[0], key.len);
+        if (v != NULL) {
+            i = (ngx_uint_t)v - 1;
+        }
+    }
```

Comment:
Because the key_max_len < 64, and which key.len >= 64, i == sizeof()/sizeof() -1, is the last one item, { ngx_null_string, 0, ngx_http_set_header }